modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java

```diff
@@ -87,6 +91,37 @@ private ResolverUtils() {
     return resolve;
   }
 
+  public static <T> T resolveValue(ValueResolver<T> resolver, ValueResolvingContext context)
+      throws MuleException {
+    T value = resolveRecursively(resolver, context);
+
+    if (value instanceof CursorProvider) {
```

Comment:
at this point you should call resolveCursor, which is the one that should have the logic to handle the TypeValue